compiler/rustc_attr/src/builtin.rs

```diff
@@ -23,8 +23,7 @@ use crate::session_diagnostics::{self, IncorrectReprFormatGenericCause};
 pub const VERSION_PLACEHOLDER: &str = "CURRENT_RUSTC_VERSION";
 
 pub fn rust_version_symbol() -> Symbol {
-    let version = option_env!("CFG_VERSION").unwrap_or("<current>");
-    let version = version.split(' ').next().unwrap();
+    let version = option_env!("CFG_RELEASE").unwrap_or("<current>");
```

Comment:
I am pretty sure this is correct, but I'm not sure how to test it ... @est31 are there any tests for this code I can run before the next release bump? I don't see anything in `tests/`.

Reply:
The function is unrelated to the release bump, the logic for that lives in `src/tools/replace-version-placeholder`, which reads the version number from `src/version`. It is ran during the release process as documented [in this PR](https://github.com/rust-lang/rust-forge/pull/643).

The `rust_version_symbol` function is used for two things: the fallback for recently stabilized features, aka ones that use `CURRENT_RUSTC_VERSION`. Tests for this are difficult because they change with the version, and change between nightly (there it's `1.17.0-dev`, or `1.17.0-nightly`) and beta+stable (there it's just `1.17.0`). I didn't really strip off the `-foo` part because I thought that it's useful to know that this is a freshly stabilized feature and it's not 100% certain yet it will arrive in that version (sometimes features that haven't made it to stable are unstabilized again, after being stabilized in nightly).

Tests are possible but it didn't come up during review so I didn't do them (e.g. in #81468 I added a test that is version-agnostic but still checks some version related code is correct, but it was for stuff that changes actual functionality, not something that changes error and documentation output). The PRs that added usages of `rust_version_symbol` this were #100591 #101772  #101215 . See the last one for a screenshot.

You can test the impact of your PR on the rustdoc by grepping for `CURRENT_RUSTC_VERSION` in library/ and then doing `./x  doc library`, and navigating to the thing that has the `CURRENT_RUSTC_VERSION`, and compare it to before your PR.

The `rust_version_symbol` function is also used for error messages, like when someone puts `#![feature(foo)]` but `foo` is already stable. I made the function to unify the code that reads the "CFG_VERSION" environment variable. I think the reason that the `the feature `{feature}` has been stable since {since} and no longer requires an attribute to enable` error is not in the testsuite is precisely because the stderr is version dependent. It should be possible to do normalize-stderr for the version number but that will then of course remove the very part that you are changing here :p.

Reply:
ok cool, looks like this is working properly :) 
![image](https://github.com/rust-lang/rust/assets/23638587/4736f92e-a49b-41a9-bf57-d75c6a404038)

> The rust_version_symbol function is also used for error messages, like when someone puts #![feature(foo)] but foo is already stable. I made the function to unify the code that reads the "CFG_VERSION" environment variable. I think the reason that the the feature {feature} has been stable since {since} and no longer requires an attribute to enable error is not in the testsuite is precisely because the stderr is version dependent. It should be possible to do normalize-stderr for the version number but that will then of course remove the very part that you are changing here :p.

I added a test that compares the output to `src/version`, which doesn't require normalization :)

Reply:
That test looks very nice!

I think you'll also need to adjust the replace-version-placeholder tool and add the test as exception. It's implemented as a very basic string replacement operation that also runs over the tests.

https://github.com/rust-lang/rust/blob/master/src/tools/replace-version-placeholder/src/main.rs#L17

Reply:
Thanks! Pushed a change that hopefully does the right thing, let me know what you think :)